### ISSUE TITLE
Village lake water is rendering at lower resolution with HD mod

### ISSUE THREAD

**Issue by RiasatSalminSami** [nipkownix/re4_tweaks#97]
![Comparison 1](https://user-images.githubusercontent.com/50959173/152983141-a43f3d06-8a2d-42c1-941b-b68078647d7f.png)

The screenshot is enough to explain the situation. It seems like the water is rendering at lower resolution. Also has bleeding issue with foreground objects. Can be noticeable with the screenshot below:
![Comparison 2](https://user-images.githubusercontent.com/50959173/152983287-14b48777-5012-4c82-a763-b13bdd9763dd.png)

This also doesn't occur with the previous 2018 version of the HD mod, regardless of GCblur option.

**Comment by Mister-Curious** [comment]
The bleeding effect I think happens no matter the HD project or not, as this is a known transparency hierarchy issue with the game engine. Can you confirm this happens with vanilla game with a  screenshot comparison?  I am not sure about the water effect rendering issue, so I cant comment on that.

**Comment by RiasatSalminSami** [comment]
> The bleeding effect I think happens no matter the HD project or not, as this is a known transparency hierarchy issue with the game engine. Can you confirm this happens with vanilla game with a screenshot comparison? I am not sure about the water effect rendering issue, so I cant comment on that.

Yeah the vanilla game or the 2018 version of the mod still shows some bleeding. But I think due to blurrier water with the final release of the mod, the bleeding ends up looking worse.  I will try to check the vanilla game to see how the bleeding looks in that.

Honestly bleeding is of least concern here, only major and noticeable problem is the water. It's definitely rendering incorrectly since later parts of the game doesn't have this issue with water [with the hd project mod].

The water is missing some of the dirt layers too on the hd project. I think it wasn't the intended look, something just went wrong.

**Comment by Mister-Curious** [comment]
You should compare the textures for each of the rooms (the one that looks good, and the one that doesn't). Each water effect uses 60 textures that play out in an animation. I'd be interested to see if the textures are the same. 

You can use [this tool](https://quackfluffy.tumblr.com/post/101263789397/resident-evil-4-uhd-texture-extrator-repacker) to extract the .pack files for the rooms. Inside you will find the water textures. 

You will also need the [LFS tool](https://residentevilmodding.boards.net/thread/15869/bio4-exe-named-ida-database) that removes the compression before using the YZ2Tool. 

So the steps would be:

1 - Locate the pack file that is associated with the room you are testing (example from above would be room r10B so pack file would be 4400010b.pack.lfs) All pack files are located in **\Resident Evil 4\BIO4\ImagePackHD**
2 - copy this file to the folder with the LFS tool. Drag and Drop the file onto re4lfs.exe. A newly generated file without the LFS extension is created. 
3 - Place this new file in a working directory with the YZ2 tool and use the unpack.bat file. This will create a folder with all the  extracted textures. 
4 - Repeat this process for the other room (one room is for the good water, the other the bad water)
5 - Compare the file sizes in Windows for the the files, as well compare the dimensions of the file. It might even be worth comparing them in a HEX editor to see if they are dissimilar. 

One theory I have is that the DOF is interfering with the rendering of the water. Each room has the ability to have the DOF to be modified independently, that is to say, each one could be different. I am not sure how Albert edited his LIT files for each room with the DOF, but I imagine they are not all the same.

**Comment by nipkownix** [comment]
Hmm, I'll ask albert about this. Maybe he has some idea of why this doesn't happen in the old release.

**Comment by RiasatSalminSami** [comment]
> For anyone unaware, looks like albert and @RiasatSalminSami talked about this issue in re4hd.com comments: https://www.re4hd.com/?p=9471#comment-197203
> 
> > Oh yes, the blurry water effect is something I added on purpose. It looked too transparent and everything under the water looked slightly blurred in the original GameCube version.
> 
> I wonder how that is actually done though, is it a trick with the texture (eg changing resolution/texture filtering or something like that), or has albert activated something to make game start blurring part of the water?

Yeah, thanks for mentioning that. I forgot to confirm it here. Albert also said he's willing to put optional files that can disable the water blur for those who prefers that.

I'm also interested in how he made the blur. Can be useful for the DOF.


I probably should change the title of this thread since it's not a bug.

**Comment by Mister-Curious** [comment]
@emoose I can ask Albert about this the next time he and I chat. There is probably a byte in the water effect (FE effect) that was used to accomplish this. I think if he applied this same byte value across all water textures in the game you should be able to create a 'fix' that restores it to the other setting that is much clearer. You would simply have to write something that looks for a particular array, and insert the value while the room is loading. I will ask him about this.

**Comment by RiasatSalminSami** [comment]
Probably have to look at the GC code. The debug menu has some DOF options. Fiddling with some of them works but few options there does absolutely nothing [at least couldn't get them to do anything].  If anyone has the GC port with the debug menu, it's worth checking if those options work there.  I suspect those might control the actual intensity of the blur.

**Comment by nipkownix** [comment]
``EnableGCBlur" was created based on GC code already (check #5 for reference). I don't think emoose missed anything related to that filter's function. There's probably something elsewhere that's supposed to control the blur though, but I have yet to see anything like that :/
